timemachine/fe/free_energy.py

```diff
@@ -794,26 +829,21 @@ def run_sims_hrex(
     warn(f"Setting numpy global random state using seed {md_params.seed}")
     np.random.seed(md_params.seed)
 
-    lambdas = [s.lamb for s in initial_states]
-    initial_replicas = [CoordsVelBox(s.x0, s.v0, s.box0) for s in initial_states]
+    def get_summed_potential_bound_impl(bps: Iterable[BoundPotential]) -> BoundGpuImplWrapper:
+        potentials = [bp.potential for bp in bps]
+        params = [bp.params for bp in bps]
+        sp = SummedPotential(potentials, params)
+        return sp.to_gpu(np.float32).bind_params_list(params)
 
-    bps = initial_states[0].potentials  # TODO: assert initial states have compatible potentials?
-    sp = SummedPotential([bp.potential for bp in bps], [bp.params for bp in bps])
-    ubp = sp.to_gpu(np.float32)
+    bound_impls = [get_summed_potential_bound_impl(initial_state.potentials) for initial_state in initial_states]
```

Comment:
Previously running larger systems (100k atoms, 48 windows) we would run out of GPU memory, and would be good to have any expectation of how large of a system with 48 windows we can fit on an A10 GPU (16GB of memory).

Note the additional changes that impact the memory since we last ran into this:
* Integrator batches up noise (10 * N doubles), we should probably change the noise to floats
* Ixn group split, now have 2-3x (solvent vs complex) the memory for hilbert sorts
* Accumulation of energies in int128

Reply:
Ah, I had thought the main culprit in running out of GPU memory had been storage for the Hilbert grid. To be clear, have we run into this since shrinking the grid to 128^3?

Reply:
I am pretty sure we had run out of GPU memory after we had changed the hilbert grid as well. We should verify again with a system of whatever size we think is a reasonable maximum, since memory usage has changed a fair bit. 

We changed the hilbert grid April 7th last year 99e8971a23f30cfa0bc6b50fbd72e7d4d308f2b9
As of 3779b1322b30c80055ab0b0b2be11eacc1209a13 on Feb 21 of this year, we had switched to only using two potentials.

Reply:
To document this in the PR directly. the current memory usage is not dominated by the hilbert curve:

3x hilbert curve grids: `d_bin_to_idx_: 3 (grids)*HILBERT_GRID_DIM^3 * 4 (bytes) = 0.024 GB`

Rather, it's the `d_ixn_atoms_` in `neighborlist.cu` when used on the `nonbonded_all_pairs` potential:

`max_ixn_count() * 4 bytes`

Where `max_ixn_count()` is:

` num_column_blocks() * num_row_blocks() * WARP_SIZE = (N//32)*(N//32)*32 = 1.25 GB`

We can probably bring this down to 1/5th - 1/10th the cost => 250MB

A typical GPU we use has 24GB of ram. Assuming a max of 48 windows, we'd still only use 12GB of ram with a 100K atom system if we make this change.